src/main/java/com/uber/cadence/internal/replay/HistoryHelper.java

```diff
@@ -216,10 +225,14 @@ public DecisionEvents next() {
             continue;
           } else if (peekedType == EventType.DecisionTaskCompleted) {
             events.next(); // consume DecisionTaskCompleted
-            nextDecisionEventId = peeked.getEventId() + 1; // +1 for next
+            nextDecisionEventId = peeked.getEventId() + 1; // +1 for next and skip over completed
```

Comment:
What about checking id of every event?
Something like
```
if (nextDecisionEventId++ != peeked.getEventId()) {
   throw...
}
```

Reply:
nextDecisionEventId is set at this point. It has no previous value. If we want to validate eventIds being in order I think we should do it in the piece of code that handles pagination (I'll add it to that CR).